src/main/java/com/kjetland/dropwizard/activemq/ActiveMQReceiverHandlerFactory.java

```diff
@@ -0,0 +1,8 @@
+package com.kjetland.dropwizard.activemq;
+
+public interface ActiveMQReceiverHandlerFactory {
+
+    <T> ActiveMQReceiverHandler<T> createActiveMQReceiverHandler(String destination, ActiveMQReceiver<T> receiver, Class<? extends T> clazz, boolean ackMessageOnException);
+
+    <T> ActiveMQReceiverHandler<T> createActiveMQReceiverHandler(String destination, ActiveMQReceiver<T> receiver, Class<? extends T> clazz, ActiveMQBaseExceptionHandler exceptionHandler);
```

Comment:
Wouldn't it be better to have these methods named registerReceiver? That way the only change really in the ActiveMQBundle would be that they return a receiver handler instead of void so it wouldn't affect those already using it.

Reply:
Of course, much better.

The design took a full circle from a complex exxtension of the bundle to a filter pattern so the balloon was a bit blown up.